### ISSUE TITLE
[consistent-type-assertions] Specifying configuration options in YAML not working as expected

### ISSUE THREAD

**Issue by xylix** [typescript-eslint/typescript-eslint#1426]
Error 
```
Error: .eslintrc.yml:
	Configuration for rule "@typescript-eslint/consistent-type-assertions" is invalid:
	Value [{"assertionStyle":"as"},{"objectLiteralTypeAssertions":"never"},{"assertionStyle":"as","objectLiteralTypeAssertions":"never"}] should NOT have more than 1 items.```

When trying to configure consistent-type-assertions with  yaml syntax for configuration items instead of an explicit map.

Expected syntax:
```YAML
  '@typescript-eslint/consistent-type-assertions':
    - error
    - assertionStyle: as
    - objectLiteralTypeAssertions: 'never'
```

Actual working syntax: 
```YAML
  '@typescript-eslint/consistent-type-assertions':
    - error
    - {assertionStyle: as, objectLiteralTypeAssertions: 'never'}
```

**Additional Info**
```bash
npm ERR! missing script: eslint
npm ERR!
npm ERR! Did you mean this?
npm ERR!     lint

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/xylix/.npm/_logs/2020-01-10T13_57_33_836Z-debug.log
[~/Code/funkly]$ npm run lint                                                                                                                                             *[master]

> funkly@0.1.0 lint /Users/xylix/Code/funkly
> eslint src/**/*.tsx --debug

  eslint:cli CLI args: [ 'src/Blockly/blockly_component.tsx', 'src/Blockly/blockly_jsx_wrappers.tsx', 'src/Blockly/config.tsx', '--debug' ] +0ms
  eslint:cli Running on files +4ms
  eslint:config-array-factory Loading JSON config file: /Users/xylix/Code/funkly/package.json +0ms
  eslint:ignore-pattern Create with: [ IgnorePattern { patterns: [ '/node_modules/*', '/bower_components/*' ], basePath: '/Users/xylix/Code/funkly', loose: false } ] +0ms
  eslint:ignore-pattern   processed: { basePath: '/Users/xylix/Code/funkly', patterns: [ '/node_modules/*', '/bower_components/*' ] } +2ms
  eslint:ignore-pattern Create with: [ IgnorePattern { patterns: [ '/node_modules/*', '/bower_components/*' ], basePath: '/Users/xylix/Code/funkly', loose: false } ] +1ms
  eslint:ignore-pattern   processed: { basePath: '/Users/xylix/Code/funkly', patterns: [ '/node_modules/*', '/bower_components/*' ] } +0ms
  eslint:file-enumerator Start to iterate files: [ 'src/Blockly/blockly_component.tsx', 'src/Blockly/blockly_jsx_wrappers.tsx', 'src/Blockly/config.tsx' ] +0ms
  eslint:file-enumerator File: /Users/xylix/Code/funkly/src/Blockly/blockly_component.tsx +1ms
  eslint:cascading-config-array-factory Load config files for /Users/xylix/Code/funkly/src/Blockly. +0ms
  eslint:cascading-config-array-factory No cache found: /Users/xylix/Code/funkly/src/Blockly. +0ms
  eslint:config-array-factory Config file not found on /Users/xylix/Code/funkly/src/Blockly +7ms
  eslint:cascading-config-array-factory No cache found: /Users/xylix/Code/funkly/src. +1ms
  eslint:config-array-factory Config file not found on /Users/xylix/Code/funkly/src +0ms
  eslint:cascading-config-array-factory No cache found: /Users/xylix/Code/funkly. +0ms
  eslint:config-array-factory Loading YAML config file: /Users/xylix/Code/funkly/.eslintrc.yml +0ms
  eslint:config-array-factory Config file found: /Users/xylix/Code/funkly/.eslintrc.yml +24ms
  eslint:config-array-factory Loading {extends:"plugin:react/recommended"} relative to /Users/xylix/Code/funkly/.eslintrc.yml +0ms
  eslint:config-array-factory Loading plugin "react" from /Users/xylix/Code/funkly/.eslintrc.yml +1ms
  eslint:config-array-factory Loaded: eslint-plugin-react@7.16.0 (/Users/xylix/Code/funkly/node_modules/eslint-plugin-react/index.js) +1ms
  eslint:config-array-factory Plugin /Users/xylix/Code/funkly/node_modules/eslint-plugin-react/index.js loaded in: 108ms +108ms
  eslint:config-array-factory Loading plugin "react" from /Users/xylix/Code/funkly/node_modules/eslint-plugin-react/index.js +1ms
  eslint:config-array-factory Loaded: eslint-plugin-react@7.16.0 (/Users/xylix/Code/funkly/node_modules/eslint-plugin-react/index.js) +0ms
  eslint:config-array-factory Plugin /Users/xylix/Code/funkly/node_modules/eslint-plugin-react/index.js loaded in: 0ms +0ms
  eslint:config-array-factory Loading {extends:"plugin:@typescript-eslint/eslint-recommended"} relative to /Users/xylix/Code/funkly/.eslintrc.yml +1ms
  eslint:config-array-factory Loading plugin "@typescript-eslint" from /Users/xylix/Code/funkly/.eslintrc.yml +0ms
  eslint:config-array-factory Loaded: @typescript-eslint/eslint-plugin@2.15.0 (/Users/xylix/Code/funkly/node_modules/@typescript-eslint/eslint-plugin/dist/index.js) +4ms
  eslint:config-array-factory Plugin /Users/xylix/Code/funkly/node_modules/@typescript-eslint/eslint-plugin/dist/index.js loaded in: 258ms +258ms
  eslint:config-array-factory Loading parser "@typescript-eslint/parser" from /Users/xylix/Code/funkly/.eslintrc.yml +2ms
  eslint:config-array-factory Loaded: @typescript-eslint/parser@2.15.0 (/Users/xylix/Code/funkly/node_modules/@typescript-eslint/parser/dist/parser.js) +1ms
  eslint:config-array-factory Loading plugin "react" from /Users/xylix/Code/funkly/.eslintrc.yml +31ms
  eslint:config-array-factory Loaded: eslint-plugin-react@7.16.0 (/Users/xylix/Code/funkly/node_modules/eslint-plugin-react/index.js) +0ms
  eslint:config-array-factory Plugin /Users/xylix/Code/funkly/node_modules/eslint-plugin-react/index.js loaded in: 0ms +0ms
  eslint:config-array-factory Loading plugin "@typescript-eslint" from /Users/xylix/Code/funkly/.eslintrc.yml +0ms
  eslint:config-array-factory Loaded: @typescript-eslint/eslint-plugin@2.15.0 (/Users/xylix/Code/funkly/node_modules/@typescript-eslint/eslint-plugin/dist/index.js) +1ms
  eslint:config-array-factory Plugin /Users/xylix/Code/funkly/node_modules/@typescript-eslint/eslint-plugin/dist/index.js loaded in: 0ms +0ms
  eslint:cascading-config-array-factory No cache found: /Users/xylix/Code. +433ms
  eslint:config-array-factory Config file not found on /Users/xylix/Code +0ms
  eslint:cascading-config-array-factory No cache found: /Users/xylix. +0ms
  eslint:cascading-config-array-factory Stop traversing because of considered root. +0ms
  eslint:rules Loading rule 'getter-return' (remaining=274) +0ms
  eslint:rules Loading rule 'no-dupe-args' (remaining=273) +0ms
  eslint:rules Loading rule 'no-dupe-keys' (remaining=272) +1ms
  eslint:rules Loading rule 'no-unreachable' (remaining=271) +0ms
  eslint:rules Loading rule 'valid-typeof' (remaining=270) +1ms
  eslint:rules Loading rule 'no-const-assign' (remaining=269) +0ms
  eslint:rules Loading rule 'no-new-symbol' (remaining=268) +1ms
  eslint:rules Loading rule 'no-this-before-super' (remaining=267) +0ms
  eslint:rules Loading rule 'no-undef' (remaining=266) +0ms
  eslint:rules Loading rule 'no-dupe-class-members' (remaining=265) +1ms
  eslint:rules Loading rule 'no-redeclare' (remaining=264) +0ms
  eslint:rules Loading rule 'indent' (remaining=263) +1ms
  eslint:rules Loading rule 'linebreak-style' (remaining=262) +10ms
  eslint:rules Loading rule 'quotes' (remaining=261) +1ms
  eslint:rules Loading rule 'semi' (remaining=260) +2ms
  eslint:rules Loading rule 'object-curly-spacing' (remaining=259) +1ms
  eslint:rules Loading rule 'object-curly-newline' (remaining=258) +2ms
  eslint:rules Loading rule 'no-console' (remaining=257) +5ms
Error: .eslintrc.yml:
	Configuration for rule "@typescript-eslint/consistent-type-assertions" is invalid:
	Value [{"assertionStyle":"as"},{"objectLiteralTypeAssertions":"never"},{"assertionStyle":"as","objectLiteralTypeAssertions":"never"}] should NOT have more than 1 items.

    at validateRuleOptions (/Users/xylix/Code/funkly/node_modules/eslint/lib/shared/config-validator.js:138:19)
    at /Users/xylix/Code/funkly/node_modules/eslint/lib/shared/config-validator.js:193:9
    at Array.forEach (<anonymous>)
    at validateRules (/Users/xylix/Code/funkly/node_modules/eslint/lib/shared/config-validator.js:190:30)
    at validateConfigArray (/Users/xylix/Code/funkly/node_modules/eslint/lib/shared/config-validator.js:337:9)
    at CascadingConfigArrayFactory._finalizeConfigArray (/Users/xylix/Code/funkly/node_modules/eslint/lib/cli-engine/cascading-config-array-factory.js:417:13)
    at CascadingConfigArrayFactory.getConfigArrayForFile (/Users/xylix/Code/funkly/node_modules/eslint/lib/cli-engine/cascading-config-array-factory.js:271:21)
    at FileEnumerator._iterateFilesWithFile (/Users/xylix/Code/funkly/node_modules/eslint/lib/cli-engine/file-enumerator.js:320:43)
    at FileEnumerator._iterateFiles (/Users/xylix/Code/funkly/node_modules/eslint/lib/cli-engine/file-enumerator.js:301:25)
    at FileEnumerator.iterateFiles (/Users/xylix/Code/funkly/node_modules/eslint/lib/cli-engine/file-enumerator.js:251:59)
npm ERR! code ELIFECYCLE
npm ERR! errno 2
npm ERR! funkly@0.1.0 lint: `eslint src/**/*.tsx --debug`
npm ERR! Exit status 2
npm ERR!
npm ERR! Failed at the funkly@0.1.0 lint script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
```

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.15.0` |
| `@typescript-eslint/parser`        | `2.15.0` |
| `TypeScript`                       | `3.7.4` |
| `ESLint`                           | `6.8.0` |
| `node`                             | `v13.6.0` |
| `npm`                              | `6.13.4` |

**Comment by bradzacher** [comment]
Unless I'm mistaken, you're using the wrong yaml syntax.
This is why I personally don't use yaml ever. YAML just gets translated to JSON under the hood, and the representation is just hard to understand IMO.

your syntax produces the following JSON, which is wrong:
```
'@typescript-eslint/consistent-type-assertions-not-working':
  - error
  - assertionStyle: as
  - objectLiteralTypeAssertions: 'never'

// converts into this incorrect object
{
"@typescript-eslint/consistent-type-assertions-not-working": [
    "error", 
    {
      "assertionStyle": "as"
    }, 
    {
      "objectLiteralTypeAssertions": "never"
    }
  ]
}
```

If you instead drop the second `-`, then it will do the correct thing
```
'@typescript-eslint/consistent-type-assertions':
  - error
  - assertionStyle: as
    objectLiteralTypeAssertions: 'never'

// converts into the correct object style
{
"@typescript-eslint/consistent-type-assertions": [
    "error", 
    {
      "objectLiteralTypeAssertions": "never", 
      "assertionStyle": "as"
    }
  ],
}
```

**Comment by xylix** [comment]
Your example still fails with 
```Error: .eslintrc.yml:
	Configuration for rule "@typescript-eslint/consistent-type-assertions" is invalid:
	Value {"assertionStyle":"as,","objectLiteralTypeAssertions":"never"} should NOT have additional properties.
	Value "as," should be equal to one of the allowed values.
	Value {"assertionStyle":"as,","objectLiteralTypeAssertions":"never"} should match exactly one schema in oneOf.```

However, I think I'll just migrate to json. The points about represented data being hard to understand in YAML are legit
